tests/func/test_import.py

```diff
@@ -92,120 +93,128 @@ def test_import_non_cached(erepo_dir, tmp_dir, dvc, scm):
-    dst = erepo.DATA_DIR + "_imported"
-    stage_file = dst + ".dvc"
+def test_pull_imported_directory_stage(tmp_dir, dvc, erepo_dir, monkeypatch):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen({"dir": {"foo": "foo content"}}, commit="create dir")
 
-    dvc_repo.imp(erepo.root_dir, src, dst)
+    dvc.imp(fspath(erepo_dir), "dir", "dir_imported")
 
-    shutil.rmtree(dst)
-    shutil.rmtree(dvc_repo.cache.local.cache_dir)
+    shutil.rmtree("dir_imported")
+    shutil.rmtree(dvc.cache.local.cache_dir)
 
-    dvc_repo.pull([stage_file])
+    dvc.pull(["dir_imported.dvc"])
 
-    assert os.path.exists(dst)
-    assert os.path.isdir(dst)
-    trees_equal(src, dst)
+    assert os.path.isdir("dir_imported")
+    trees_equal(fspath(erepo_dir / "dir"), "dir_imported")
 
 
-def test_download_error_pulling_imported_stage(dvc_repo, erepo):
-    src = erepo.FOO
-    dst = erepo.FOO + "_imported"
+def test_download_error_pulling_imported_stage(
+    tmp_dir, dvc, erepo_dir, monkeypatch
+):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen("foo", "foo content", commit="create foo")
+    dvc.imp(fspath(erepo_dir), "foo", "foo_imported")
 
-    dvc_repo.imp(erepo.root_dir, src, dst)
-
-    dst_stage = Stage.load(dvc_repo, "foo_imported.dvc")
+    dst_stage = Stage.load(dvc, "foo_imported.dvc")
     dst_cache = dst_stage.outs[0].cache_path
 
-    os.remove(dst)
+    os.remove("foo_imported")
     os.remove(dst_cache)
 
     with patch(
         "dvc.remote.RemoteLOCAL._download", side_effect=Exception
     ), pytest.raises(DownloadError):
-        dvc_repo.pull(["foo_imported.dvc"])
+        dvc.pull(["foo_imported.dvc"])
 
 
 @pytest.mark.parametrize("dname", [".", "dir", "dir/subdir"])
-def test_import_to_dir(dname, dvc_repo, erepo):
-    src = erepo.FOO
-
+def test_import_to_dir(dname, tmp_dir, dvc, erepo_dir, monkeypatch):
     makedirs(dname, exist_ok=True)
 
-    stage = dvc_repo.imp(erepo.root_dir, src, dname)
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen("foo", "foo content", commit="create foo")
+
+    stage = dvc.imp(fspath(erepo_dir), "foo", dname)
 
-    dst = os.path.join(dname, os.path.basename(src))
+    dst = os.path.join(dname, "foo")
 
     assert stage.outs[0].fspath == os.path.abspath(dst)
     assert os.path.isdir(dname)
-    assert filecmp.cmp(erepo.FOO, dst, shallow=False)
+    assert (tmp_dir / dst).read_text() == "foo content"
 
 
-def test_pull_non_workspace(git, dvc_repo, erepo):
-    src = "version"
-    dst = src
+def test_pull_non_workspace(tmp_dir, scm, dvc, erepo_dir, monkeypatch):
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc_gen("foo", "master content", commit="create foo")
+        erepo_dir.scm.checkout("new_branch", create_new=True)
+        erepo_dir.dvc_gen("foo", "branch content", commit="modify foo")
```

Comment:
This is a bit complicated bootstrapping. We will need to come up with something better.